airflow/configuration.py

```diff
@@ -1531,14 +1531,28 @@ def initialize_config() -> AirflowConfigParser:
         if local_conf.getboolean("core", "unit_test_mode"):
             local_conf.load_test_config()
 
-    # Make it no longer a proxy variable, just set it to an actual string
+    old_webserver_config = AIRFLOW_HOME + "/webserver_config.py"
+    # Prioritise airflow webserver config that is present in the config location
+    webserver_config_in_config = AIRFLOW_HOME + "/config/webserver_config.py"
+
     global WEBSERVER_CONFIG
-    WEBSERVER_CONFIG = AIRFLOW_HOME + "/webserver_config.py"
+    WEBSERVER_CONFIG = webserver_config_in_config
+
+    msg = (
+        "webserver_config.py is still present in {}, this location is deprecated. You should move your "
+        "webserver_config.py to {} "
+    )
+
+    msg = msg.format(old_webserver_config, webserver_config_in_config)
+
+    if os.path.isfile(old_webserver_config):
+        warnings.warn(msg, category=DeprecationWarning)
 
     if not os.path.isfile(WEBSERVER_CONFIG):
         import shutil
 
         log.info("Creating new FAB webserver config file in: %s", WEBSERVER_CONFIG)
+        os.makedirs(os.path.dirname(AIRFLOW_HOME + "/config"), exist_ok=True)
```

Comment:
Isn’t this just `os.makedirs(AIRFLOW_HOME, exist_ok=True)`?

Reply:
Thanks for the feedback. I dont think so. It is looking for a directory within `AIRFLOW_HOME` and not just for `AIRFLOW_HOME`. Maybe is would be equal to `os.makedirs(AIRFLOW_HOME + "/config", exist_ok=True)`

Did I understand your question right?

Reply:
`os.path.dirname(AIRFLOW_HOME + "/config")` returns _the directory `config` is in_, which is simply `AIRFLOW_HOME`. Maybe `dirname` is not what you think and should not be used here?